includes/modules/shipping_estimator.php

```diff
@@ -252,24 +252,24 @@ function shipincart_submit(){
         $show_in = FILENAME_SHOPPING_CART;
     }
     if (zen_is_logged_in() && !zen_in_guest_checkout()) {
-        $addresses = $db->execute("SELECT address_book_id, entry_city AS city, entry_postcode AS postcode, entry_state AS state, entry_zone_id AS zone_id, entry_country_id AS country_id FROM " . TABLE_ADDRESS_BOOK . " WHERE customers_id = '" . (int)$_SESSION['customer_id'] . "'");
+        $addresses = $db->Execute("SELECT address_book_id, entry_city AS city, entry_postcode AS postcode, entry_state AS state, entry_zone_id AS zone_id, entry_country_id AS country_id FROM " . TABLE_ADDRESS_BOOK . " WHERE customers_id = '" . (int)$_SESSION['customer_id'] . "'");
```

Comment:
#4942 
i am of the opinion, that this code should be:
```
$customer = new Customer($_SESSION['customer_id']);
```
`$addresses` would then be `$addresses = $customer->getAddressBookEntries();`

and then you would have spotted the bug listed above.

if we have a class, **lets use it!**

Reply:
Sure, once it is clarified.

Reply:
@proseLA PR merged.

Reply:
I was waiting for this PR to be cleared/squashed as this class use is a real change.

Reply:
This/my branch does not incorporate the class change and although I have attempted to bring in the changes in the z158 branch subsequently, I am not convinced/confident of the commit results.

Reply:
The PR #4949 was merged, and the function getAddressBookEntries returns an object. 
Check includes/classes/Customer.php in an updated version of the Zen Cart 158 branch.

Reply:
Sure, but I don't have clear how to get that change into this branch.
But that is academic, I've just started to try this and it is not so straightforward as expected, it is not drop-in, so should definately be in a separate PR.

Reply:
is anything so straight forward?